### PR TITLE
Migrate to ui.router, fixes 113

### LINKED ISSUE #113: Migrate to ui.router
Current behaviour:
We're using [angular-router](https://docs.angularjs.org/api/ngRoute/service/$route) which is a legacy router.

Expected behaviour:
 - Get rid of `data-ng-controller` directives and use router to define controllers
 - Use [Angular-ui-router](https://github.com/angular-ui/ui-router). 
 - Add ControllerAs functionality
 - Use defined functions instead of lambda function as constructor

### PR DESCRIPTION
**Pull request by alihaghighatkhah**
Fixes #113

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ module.exports = function (grunt) {
             angular: {
                 src: ['bower_components/angular/angular.min.js',
                       'bower_components/angular-resource/angular-resource.min.js',
-                      'bower_components/angular-route/angular-route.min.js',
+                      'node_modules/angular-ui-router/release/angular-ui-router.min.js',
```
Comment (karmacoma) [inline Gruntfile.js]: Is it possible to install `angular-ui-router` as a bower component, like the other angular dependencies?

File: public/src/js/services/marketWatcher.js [hunk 2]
```diff
@@ -77,66 +78,68 @@ var MarketWatcher = function ($q, $http, $scope) {
...
     });
 };
 
 angular.module('lisk_explorer.tools').factory('marketWatcher',
-  function ($q, $http, $socket) {
-      return function ($scope) {
-          var marketWatcher = new MarketWatcher($q, $http, $scope),
+  function ($q, $http, $socket, $rootScope) {
+      return function (vm) {
+          console.log('passing', vm);
```
Comment (Isabello) [inline public/src/js/services/marketWatcher.js]: Can we squelch these console logs?

File: Gruntfile.js (hunk 1 above)
Comment (slaweet) [inline Gruntfile.js]: What about installing all dependencies by npm?
Webpack will be happier once we use it: https://webpack.github.io/docs/usage-with-bower.html#prefer-modules-from-npm-over-bower

Comment (alihaghighatkhah) [inline Gruntfile.js]: As they've mentioned [here](https://ui-router.github.io/ng1/):

> bower releases are considered ‘legacy’

So I decided to use the npm installer.

Comment (Isabello) [inline Gruntfile.js]: Is there a way to load this through bower instead of node modules? Just a consistency thing. Its fine other wise

Comment (karmacoma) [inline Gruntfile.js]: @alihaghighatkhah Okay understood 👍 

@slasheks If this is possible, then sure, but not in this pull request.

Comment (Isabello) [inline Gruntfile.js]: I am glad im not the only one who asked about this. I definitely would prefer node modules over bower as well.

**Pull request merged**